### PR TITLE
feat(schema): add Validators library with core rule set and full unit tests

### PR DESCRIPTION
**Pull request by MichaelMcKibbin**
Validators subsystem for the schema/typing layer.

Provides a composable, extensible validation framework that can be applied to individual fields or entire rows during CSV schema mapping.

Includes:

A functional Validator<T> interface
A full library of reusable validation rules
Rule composition (AND / OR / NOT)
A helper for lenient error collection (validateAll)
A complete JUnit test suite covering behaviour and edge cases

Lays the foundation for upcoming Schema, FieldSpec, RowBuilder, and ValidationError integration.

**Validators class**

General rules
required()
custom(predicate, message)

String rules
length(min, max)
regex(pattern)
enumOf(...)

Numeric rules
min(BigDecimal)
max(BigDecimal)

Composition
and(...)
or(...)
not(...)

Utility
validateAll(value, validators, errors) for error aggregation

Note: 
Non-required validators ignore null to allow predictable composition.
Fail-fast behaviour when used individually.
Lenient error collection when used with validateAll.

**ValidatorsTest**

JUnit 5 test suite verifying:

- Required field behaviour (null/blank rejection)
- String validation: length, regex, enum membership
- BigDecimal validation: min/max bounds
- Custom predicate validation
- Composition correctness for and, or, and not
- Inversion logic (not) with corrected implementation
- Error collection with validateAll
- 

All tests pass.

Adding this now allows forward progress on:

- Schema definition structures
- FieldSpec (type + validators)
- RowBuilder typed mapping
- Integration with DateTimeSpec and DecimalSpec
- ValidationError and error-cap logic

image

### CODE REVIEW

File: src/main/java/com/group5/csv/schema/Validators.java [hunk 1]
```diff
@@ -1,2 +1,299 @@
...
+        };
+    }
+
+    /**
+     * String length validator: inclusive [min, max].
+     * Null values are ignored (can be forced by using required()).
+     */
+    public static Validator<String> length(int min, int max) {
+        if (min < 0 || max < min) {
+            throw new IllegalArgumentException("Invalid length boundries. Limits are: min=" + min + ", max=" + max);
```
Comment (edsonesf) [inline src/main/java/com/group5/csv/schema/Validators.java]: Minor: typo `boundries`

**Comment by MichaelMcKibbin** [comment]
Is it OK to merge this now?

**Comment by MichaelMcKibbin** [comment]
rechecked that pull from main done first :)
